### PR TITLE
Mass email script

### PR DESCRIPTION
**Pull request by nicholaspad**
## Goal
- Create an email script that can send HTML-formatted (e.g. ``, `<b>`, `<a>`) emails to all TigerSnatch users.
    - Also provide the ability to send a "test" email to only admins, in order to check formatting etc.
- Use a SendGrid template to make mass emails look nicer.
- Remove the need to use Outlook in combination with manually getting chunks of 500 users to send TigerSnatch emails.

## Functionality
- Use `--test` to send the email to only admins as listed in the admin collection.
- Use `--all` to send the email to all users.

## Testing
- Set the database URL to the dev database to avoid emailing 2500 people.
- Create files `SUBJECT` and `MESSAGE` in the same directory as the script. Type an email subject in `SUBJECT` and an email body in `MESSAGE`.
- Run `python src/_email_all_users.py --test` and follow the prompts. Check your email (if you're an admin) to verify it's how you want.
- Run `python src/_email_all_users.py --all` and follow the prompts.

## Example run
```
$ python src/_email_all_users.py --all
---------------
SUBJECT: Your subject goes here!

MESSAGE:
Your <b><a href="https://tigersnatch.com">message</a></b> goes here!

You can use multiple lines if necessary.
---------------

Send the above email to ALL users (2 in total)? (y/n) y
Sending... 2 emails sent!
```

This is what emails to only admins (`--test` flag) look like (doesn't have a hyperlink in it because it's an older screenshot):

image

This is what emails to all users (`--all` flag) look like:

image

### CODE REVIEW

**Review by shannon-heh — COMMENTED** [review]
what do you think about creating a separate file where a dev can write the SUBJECT and MESSAGE, and the script reads it in? This way, we avoid the risk of someone committing their custom inputs and the script code never changes. We can add the name of this file to the .gitignore too for extra precaution. @nicholaspad

File: src/_email_all_users.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# ----------------------------------------------------------------------
+# _email_all_users.py
+# Note: It is best to run this script *on your local computer*. You must
+# make sure that all environment variables are already set. See Heroku
+# Config Vars to get these variables.
+#
+# Script to email all TigerSnatch users. This uses a SendGrid dynamic
+# template (edit on SendGrid --> Dynamic Templates). To use this script:
+#
+# 1. Create a file named MESSAGE in the same directory as this script
```
Comment (shannon-heh) [inline src/_email_all_users.py:10]: does the file type matter?

File: src/_email_all_users.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from sys import exit, argv
+from sendgrid import SendGridAPIClient
+from config import SENDGRID_API_KEY, TS_EMAIL
+
+if __name__ == "__main__":
+
+    def process_args():
+        if len(argv) != 2 or (argv[1] != "--test" and argv[1] != "--all"):
+            print("specify one of the following flags:")
+            print("\t--test: send email to only admins")
```
Comment (shannon-heh) [inline src/_email_all_users.py:36]: for `--test`, is there a specific reason we're sending an email to all admins. is it possible to only send to yourself? can you specify an argument in addition to `--test` that is your own email?

Comment (shannon-heh) [reply src/_email_all_users.py:36]: sorry im being picky, i just wanna make sure this script is really easy to use

File: src/_email_all_users.py (hunk 1 above)
Comment (nicholaspad) [reply src/_email_all_users.py:10]: there is no filetype. the file should be called simply `MESSAGE` or `SUBJECT` with no extension

File: src/_email_all_users.py (hunk 2 above)
Comment (nicholaspad) [reply src/_email_all_users.py:36]: yep, good idea. i'll just make it so `--test` requires 1 or more email addresses

File: src/_email_all_users.py (hunk 1 above)
Comment (nicholaspad) [reply src/_email_all_users.py:10]: i'll clarify that in the instructions

**Review by shannon-heh — APPROVED** [review]
yay!!! makes sending emails so much easier for devs and visually pleasing for users!

**Pull request merged**